### PR TITLE
feat: DCC homepage refresh

### PR DESCRIPTION
**Pull request by mamieorine**
Changes proposed in this pull request:

- Developed CSS for the homepage refresh
- Added new assets to the project
- Enhanced gulp script to be able to copy /static/dcc/* to the /dist folder.

### CODE REVIEW

**Review by matthiasrohmer — COMMENTED** [review]
That doesn't look too bad. I'll follow up via e-mail too to share some more details.

image

- Cards stick to the edge of the screen

- Please also add the Open Sans font to both the assets as well as the CSS. While we wait for approval for Android Euclid use Poppins (https://fonts.google.com/specimen/Poppins, Bold 600/700 maybe) as a stand-in.

- hero.svg doesn't gracefully scale on mobile screens

- You can also use this PR to add a second header which is then only used for the homepage. Implement it in a Nunjucks partial that we can then include - we'll don't need the 'Subscribe' button, avatar and menu button from the Figma but everything else.

File: site/_static/static/dcc/hero.svg
Comment (matthiasrohmer) [inline site/_static/static/dcc/hero.svg:1]: This shouldn't be an SVG as base64 encoding the image makes it considerably bigger than what the original image would be.

File: site/_static/static/dcc/open-icon.svg
Comment (matthiasrohmer) [inline site/_static/static/dcc/open-icon.svg:1]: Same as with hero.svg - this is a raster image actually and should be exported as one, so that rather be a PNG or something.

File: site/_static/static/dcc/powerful-icon.svg
Comment (matthiasrohmer) [inline site/_static/static/dcc/powerful-icon.svg:1]: Same as with hero.svg - this is a raster image actually and should be exported as one, so that rather be a PNG or something.

File: site/_static/static/dcc/solutions-icon.svg
Comment (matthiasrohmer) [inline site/_static/static/dcc/solutions-icon.svg:1]: Same as with hero.svg - this is a raster image actually and should be exported as one, so that rather be a PNG or something.

**Review by matthiasrohmer — COMMENTED** [review]
- Please remove the sidebar, it's redundant to the updated header
image

- Add articles to the updated header as it would be inaccessible when we remove the sidebar

- Please add `:hover` states to the header nav items

- The search button dangles in the void for some narrow viewports
image

- When search is opened on narrow viewports it weirdly breaks
image

- Click on the burger menu doesn't do anything for mobile

- Please check the horizontal rhythm in the header. I know it's not perfect in the Figma either, but the typo upper boundaries should match, respectively the Developers `s` should always be as big as the `G` of Get inspired.

Screenshot 2023-08-25 at 18 10 46

- There is a serif font flash before the web fonts are loaded

File: site/_includes/layouts/devsite.njk
Comment (devnook) [reply site/_includes/layouts/devsite.njk:19]: afaik they will be loaded from googleapis anyways on Devsite, so I'd keep it simple here. We should add preconnect though link rel="preconnect"

File: site/_includes/partials/devsite-header.njk
Comment (devnook) [inline site/_includes/partials/devsite-header.njk:25]: Please no untraslatable strings in template

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [inline site/_scss/layouts/devsite-theme.scss:111]: use --devsite-buttons-padding

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [inline site/_scss/layouts/devsite-theme.scss:110]: --devsite-buttons-margin

**Review by devnook — CHANGES_REQUESTED** [review]
It seems you are targeting a lot of devsite- classes there, which is not allowed. See the instructions doc.

File: site/_includes/partials/devsite-header.njk
Comment (mamieorine) [reply site/_includes/partials/devsite-header.njk]: Hi Ewa, sorry we don't quite understand what this means. Where are the no untranslatable strings in the lines that you have requested change?

File: site/_includes/layouts/devsite.njk
Comment (devnook) [reply site/_includes/layouts/devsite.njk:19]: Please address this comment.

Comment (mamieorine) [reply site/_includes/layouts/devsite.njk:21]: Hi Ewa, @simgobb  has added `rel="preconnect"` already. 
```
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```

Please re-review, thank you 😊

File: site/_includes/partials/devsite-header.njk
Comment (devnook) [inline site/_includes/partials/devsite-header.njk:33]: These strings will not get translated if you hardcode them. Please use i18n filter as in the example above.

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [inline site/_scss/layouts/devsite-theme.scss:123]: These paddings seem too big (e.g. bottom 68px vs 39 in Figma)

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [inline site/_scss/layouts/devsite-theme.scss:126]: I assume you target a card label here. Please use the specific selector, rather than every span on the card (yes, this is this place where devsite- selector makes more sense than "span")

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [inline site/_scss/layouts/devsite-theme.scss:159]: Remove whitespace

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [inline site/_scss/layouts/devsite-theme.scss:218]: Please make it a universal class, that can be also used outside of services card (like .dcc-icon-button). Adapt dcc-web-app-features-link calls below as well.

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [inline site/_scss/layouts/devsite-theme.scss:540]: Whitespace.

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [inline site/_scss/layouts/devsite-theme.scss:559]: Whitespace.

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [inline site/_scss/layouts/devsite-theme.scss:668]: Since these styles do not influence the look and feel of the content from devsite, please remove to another file.

**Review by devnook — CHANGES_REQUESTED** [review]
Reviewed

File: site/_includes/partials/devsite-navigation-rail.njk
Comment (mamieorine) [reply site/_includes/partials/devsite-navigation-rail.njk:2]: There are 2 places different from the navigation-rail.njk file, the logo and the rail items `{% for name, item in nav.devsite_rail %}` (please see the [latest commit ](https://github.com/GoogleChrome/developer.chrome.com/pull/7080/commits/a04df0e7394fd2c3381eecb7b1456b98aa83a485)).

_**Please remove duplicated code and just replace what differs**_ - I am not sure what approach we should take, do you want us to use the old file and add the condition checking the new template?

Comment (devnook) [reply site/_includes/partials/devsite-navigation-rail.njk:2]: Any solution that get you the desired outcome and does not involve repeating 134 lines of code.

Comment (mamieorine) [reply site/_includes/partials/devsite-navigation-rail.njk]: Thank you for clarifying, I have fixed this, please see the latest commit -https://github.com/GoogleChrome/developer.chrome.com/pull/7080/commits/55105f1fa1e560a8c23798db4cadd309bdcf4256.

File: site/_includes/partials/devsite-header.njk
Comment (matthiasrohmer) [inline site/_includes/partials/devsite-header.njk:1]: I think if we use the new header across all of the site we should also get rid of the sidebar everywhere and use the exact same header on every page. This would also fix the content jump when you navigate away from the homepage, as the other header is slightly smaller.

**Review by matthiasrohmer — COMMENTED** [review]
Quite a few images are still SVGs, why they are actually raster images.

- site/_static/static/dcc/bringing-safety-thumb.svg
- site/_static/static/dcc/case-study-thumb.svg
- site/_static/static/dcc/chrome-111-thumb.svg
- site/_static/static/dcc/chrome-112-thumb.svg
- site/_static/static/dcc/chrome-dev-insider-thumb.svg

File: site/_includes/layouts/devsite.njk
Comment (devnook) [reply site/_includes/layouts/devsite.njk:127]: This is still happening:
Screenshot 2023-09-07 at 15 47 47

File: site/_includes/layouts/devsite.njk
Comment (devnook) [inline site/_includes/layouts/devsite.njk:132]: This script prevents from the anchor to get attached in the url bar.
Please use css rule scroll-behavior: smooth; instead and reove unnecessary js.

File: site/_scss/layouts/devsite-theme.scss
Comment (devnook) [reply site/_scss/layouts/devsite-theme.scss:196]: Was this addressed?

File: site/_scss/_global.scss
Comment (matthiasrohmer) [inline site/_scss/_global.scss:34]: Can we please rather set those on `body`? Using `:root` for actual properties and not only var declarations seems weird.

File: site/_scss/globals/_fonts.scss
Comment (matthiasrohmer) [inline site/_scss/globals/_fonts.scss:158]: Good to have this added, but please move to devsite-theme instead so that they do not get loaded on pages that are not the homepage, where we do not use those icons yet. Please correct me if I am wrong and they are required for other pages as well.

File: site/_scss/layouts/devsite-theme.scss
Comment (mamieorine) [reply site/_scss/layouts/devsite-theme.scss:196]: Sorry, I did not explain why we did it that way. 

All `dcc-blog-card` cards have the label on top so I set the position of the label to **_absolute_** because the label is wrapped in the description element and the description in some cards is displayed below the image. I cannot think of other ways to move this label up by using CSS except to set the position to absolute. To avoid overlapping between the label and heading, I need to add extra padding-top into it. The layout of the blog card looks good to me, please see the screenshot below:

Screenshot 2023-09-07 at 15 16 53

Comment (mamieorine) [reply site/_scss/layouts/devsite-theme.scss:196]: In my opinion, the better way is to adjust the HTML element by moving the label out of the description element and place on top of the card. As you see in the screenshot below, the label is in `devsite-landing-row-item-description` element. 
Screenshot 2023-08-24 at 15 35 33.

Please let me know if there is another better way. I will fix this ASAP. Thank you!

File: site/_scss/globals/_fonts.scss
Comment (mamieorine) [reply site/_scss/globals/_fonts.scss]: I moved this to the `devsite-theme.scss` already.

File: site/_scss/_global.scss
Comment (mamieorine) [reply site/_scss/_global.scss]: I have fixed this in [this commit](https://github.com/GoogleChrome/developer.chrome.com/pull/7080/commits/f943fb03a2d1af33ef021e0f8fb58d14606fce77).

File: site/_scss/layouts/devsite-theme.scss
Comment (matthiasrohmer) [reply site/_scss/layouts/devsite-theme.scss:243]: As altering the HTML is not an option I agree that this is the least invasive solution without heavily altering the surrounding layout logic, thanks for explaining.

File: site/_scss/layouts/devsite-theme.scss
Comment (matthiasrohmer) [reply site/_scss/layouts/devsite-theme.scss:246]: @mamieorine, please target `.devsite-landing-row-item-labels` (the parent element of the span you are currently targeting) instead to avoid accidentally affecting any spans that might appear in that element in the future.

File: site/_scss/layouts/devsite-theme.scss
Comment (matthiasrohmer) [reply site/_scss/layouts/devsite-theme.scss:596]: @simgobb, @mamieorine: that's a double new-line which can be condensed to one.

File: site/_includes/layouts/devsite.njk
Comment (mamieorine) [reply site/_includes/layouts/devsite.njk]: I have fixed this. 
Screenshot 2023-09-07 at 15 43 28

File: site/_includes/partials/devsite-side-nav.njk
Comment (mamieorine) [reply site/_includes/partials/devsite-side-nav.njk]: This file has been removed. I replaced the current header with the new (devsite) header so now the new header and the new side nav are applied globally.

File: site/_includes/partials/devsite-footer.njk
Comment (mamieorine) [reply site/_includes/partials/devsite-footer.njk]: This issue has been fixed. I added this condition `{% if not isDevsite %}` to check if the page is the homepage, if so just remove the social row.

File: site/_scss/layouts/devsite-theme.scss
Comment (mamieorine) [reply site/_scss/layouts/devsite-theme.scss:246]: @matthiasrohmer  I think you might reviewed the old file, we updated this selector a few days back. Please see the latest file [here](https://github.com/GoogleChrome/developer.chrome.com/blob/4956d82e2973630b36fd268f5e720413ce6baa7c/site/_scss/layouts/devsite-theme.scss. ). Thank you 😊

File: site/_scss/layouts/devsite-theme.scss
Comment (mamieorine) [reply site/_scss/layouts/devsite-theme.scss]: @matthiasrohmer  I think you might reviewed the old file, we added `.dcc-icon-button` for each link and added style to this selector a few days back. Please see the [latest file here](https://github.com/GoogleChrome/developer.chrome.com/blob/4956d82e2973630b36fd268f5e720413ce6baa7c/site/_scss/layouts/devsite-theme.scss. ). Thank you 😊

File: site/_scss/layouts/devsite-theme.scss
Comment (mamieorine) [reply site/_scss/layouts/devsite-theme.scss:596]: We removed this double new line. Please see the [latest file here](https://github.com/GoogleChrome/developer.chrome.com/blob/4956d82e2973630b36fd268f5e720413ce6baa7c/site/_scss/layouts/devsite-theme.scss.). Thank you 😊

File: site/_scss/layouts/devsite-theme.scss
Comment (mamieorine) [reply site/_scss/layouts/devsite-theme.scss]: Removed

File: site/_scss/layouts/devsite-theme.scss
Comment (mamieorine) [reply site/_scss/layouts/devsite-theme.scss]: We moved all code in `/* Layout, header, and footer */ ` out, and replaced the current layout with the devsite layout following Matthias' feedback so the whole site now has the new header and the new side nav for mobile.

File: site/_scss/layouts/devsite-theme.scss
Comment (mamieorine) [reply site/_scss/layouts/devsite-theme.scss:279]: removed 😊

**Comment by matthiasrohmer** [comment]
Merged in https://github.com/GoogleChrome/developer.chrome.com/pull/7302.

**Pull request closed without merging**
